genie-web/src/main/java/com/netflix/genie/web/configs/MvcConfig.java

```diff
@@ -86,14 +86,22 @@ public String hostName() throws UnknownHostException {
     }
 
     /**
-     * Get an HttpClient for calling between Genie nodes.
+     * Get RestTemplate for calling between Genie nodes.
      *
-     * @return The http client to use
+     * @param httpConnectTimeout http connection timeout in milliseconds
+     * @param httpReadTimeout http read timeout in milliseconds
+     * @return The rest template to use
      */
     @Bean
     @ConditionalOnMissingBean
```

Comment:
Did you verify that no where else in the spring boot auto configuration stack another rest template wasn't being created with this name that would cause yours to not be instantiated due to this conditional? I ask cause it seems risky as we have specific things in here we want to be setup and we can't guarantee another instance has it. May also want to use qualifiers where it is wired in.